exp/controllers/ocimanaged_machinepool_controller.go

```diff
@@ -117,7 +117,7 @@ func (r *OCIManagedMachinePoolReconciler) Reconcile(ctx context.Context, req ctr
 	ociManagedCluster := &infrastructurev1beta2.OCIManagedCluster{}
 	ociClusterName := client.ObjectKey{
 		Namespace: cluster.Namespace,
-		Name:      cluster.Name,
+		Name:      cluster.Spec.InfrastructureRef.Name,
```

Comment:
this is the actual issue we fixed, the problem is that we were assuming cluster name == infra cluster name which is not true. Infra cluster name should be looked up via spec, we were already doing it in machine controller.